packages/flutter/lib/src/material/dropdown_menu.dart

```diff
@@ -613,14 +613,23 @@ class _DropdownMenuState<T> extends State<DropdownMenu<T>> {
       .toList();
   }
 
+  bool _shouldUpdateCurrentHighlight(List<DropdownMenuEntry<T>> entries, TextEditingController textEditingController) {
+    final String searchText = textEditingController.value.text.toLowerCase();
+    if (searchText.isEmpty) {
+      return true;
+    }
+    if (currentHighlight != null && currentHighlight! < entries.length && entries[currentHighlight!].label.toLowerCase().contains(searchText)) {
+      return false;
+    }
+    return true;
+  }
+
   int? search(List<DropdownMenuEntry<T>> entries, TextEditingController textEditingController) {
     final String searchText = textEditingController.value.text.toLowerCase();
     if (searchText.isEmpty) {
       return null;
     }
-    if (currentHighlight != null && entries[currentHighlight!].label.toLowerCase().contains(searchText)) {
```

Comment:
Sorry for the late response! Seems just adding `currentHighlight < entries.length` in this if statement can fix the problem right?

Reply:
Yes, but we are also separating the logic for `highlight` visibility with current change, WDYT should we just add `currentHighlight < entries.length` instead of separation of logic ?